linker/Tests/TestCasesRunner/TestCaseMetadaProvider.cs

```diff
@@ -79,5 +101,17 @@ T GetOptionAttributeValue<T> (string attributeName, T defaultValue)
 
 			return defaultValue;
 		}
+
+		private CompileAssemblyInfo CreateCompileAssemblyInfo (CustomAttribute attribute)
+		{
+			return new CompileAssemblyInfo
+			{
+				OutputName = (string) attribute.ConstructorArguments [0].Value,
+				SourceFiles = ((CustomAttributeArgument []) attribute.ConstructorArguments [1].Value).Select (arg => _testCase.SourceFile.Parent.Combine (arg.Value.ToString ())).ToArray (),
```

Comment:
This is very ugly, at least store attribute.ConstructorArguments in local variable